draft-kinnear-webtransport-http2.md

```diff
@@ -63,459 +78,248 @@ Discussion of this draft takes place on the WebTransport mailing list
-is not intended for data exchange.
+After the session is established, the peers can exchange data using the
+following mechanisms:
 
+* Both client and server can create a bidirectional or unidirectional stream
+  using a new HTTP/2 extension frame (WT_STREAM)
+* A datagram can be sent using a a new HTTP/2 extension frame WT_DATAGRAM.
 
-## WebTransport Streams
+An WebTransport session is terminated when the CONNECT stream that created it
+is closed.
 
-Following the establishment of a WebTransport Connect stream, either the client
-or the server can initiate a WebTransport Stream by sending the WTHEADERS frame,
-defined in {{the-wtheaders-frame}}. This frame references an open WebTransport
-Connect stream which is used by any intermediaries to correctly forward the
-stream to the destination endpoint. The only frames allowed on WebTransport
-Streams are WTHEADERS, CONTINUATION, DATA and any negotiated extension frames.
+# Session Establishment
 
+## Establishing a Transport-Capable HTTP/2 Connection
 
-## Negotiation
+In order to indicate support for WebTransport, both the client and the server
+MUST send a SETTINGS_ENABLE_WEBTRANSPORT value set to "1" in their SETTINGS
+frame.  Endpoints MUST NOT use any WebTransport-related functionality unless
+the parameter has been negotiated.
 
-Clients negotiate the use of WebTransport over HTTP/2 using both the SETTINGS
-frame and one or more extended CONNECT requests as defined in {{!RFC8441}}.
+## Extended CONNECT in HTTP/2
 
-Use of the extended CONNECT method extension requires the
-SETTINGS_ENABLE_CONNECT_PROTOCOL parameter to be received by a client prior to
-its use. An endpoint that supports receiving the extended CONNECT method SHOULD
-send this setting with a value of 1.
+{{!RFC8441}} defines an extended CONNECT method in Section 4, enabled by the
+SETTINGS_ENABLE_CONNECT_PROTOCOL parameter.  An endpoint doesn not need to send
+both SETTINGS_ENABLE_CONNECT_PROTOCOL and SETTINGS_ENABLE_WEBTRANSPORT; the
+SETTINGS_ENABLE_WEBTRANSPORT setting implies that an endpoint supports extended
+CONNECT.
 
-The extended CONNECT method extension uses the `:protocol` pseudo-header field
-to negotiate the protocol that will be used on a given stream in an HTTP/2
-connection. This document registers a new token, "webtransport", in the
-"Hypertext Transfer Protocol (HTTP) Upgrade Token Registry" established by
-{{?RFC7230}} and located at
-<https://www.iana.org/assignments/http-upgrade-tokens/>.
+## Creating a New Session
 
-This token is used in the `:protocol` pseudo-header field to indicate that the
-endpoint wishes to use the WebTransport protocol on the new stream.
+As WebTransport sessions are established over HTTP/2, they are identified
+using the `https` URI scheme {{!RFC7230}}.
 
+In order to create a new WebTransport session, a client can send an HTTP
+CONNECT request.  The `:protocol` pseudo-header field ({{!RFC8441}}) MUST be
+set to `webtransport`.  The `:scheme` field MUST be `https`.  Both the
```

Comment:
Previously, this document registered that value. If that's now in WT over HTTP/3, we probably should reference that here?